File: SOLID/C#/SRP/OrderProcessor.cs
public class OrderProcessor
{
    private readonly PricingService pricing;
    private readonly DiscountService discount;
    private readonly OrderRepository repository;
    private readonly EmailNotificationService notifier;
    private readonly AuditLogger audit;

    public OrderProcessor(
        PricingService pricing,
        DiscountService discount,
        OrderRepository repository,
        EmailNotificationService notifier,
        AuditLogger audit
    )
    {
        this.pricing = pricing;
        this.discount = discount;
        this.repository = repository;
        this.notifier = notifier;
        this.audit = audit;
    }

    public void Process(string orderId, double amount, string email)
    {
        double taxedAmount = pricing.ApplyTax(amount);
        double discountValue = discount.ApplyDiscount(amount);
        double finalAmount = taxedAmount - discountValue;

        repository.Save(orderId, finalAmount);
        notifier.Notify(email);
        audit.Log($"Order processed successfully: {orderId}");
    }
}

File: SOLID/Java/SRP/OrderProcessor.java
package SRP;

class OrderProcessor {

    private final PricingService pricingService;
    private final DiscountService discountService;
    private final OrderRepository repository;
    private final EmailNotificationService notifier;
    private final AuditLogger auditLogger;

    public OrderProcessor(
            PricingService pricingService,
            DiscountService discountService,
            OrderRepository repository,
            EmailNotificationService notifier,
            AuditLogger auditLogger
    ) {
        this.pricingService = pricingService;
        this.discountService = discountService;
        this.repository = repository;
        this.notifier = notifier;
        this.auditLogger = auditLogger;
    }

    public void process(String orderId, double amount, String email) {
        double taxedAmount = pricingService.applyTax(amount);
        double discount = discountService.applyDiscount(amount);
        double finalAmount = taxedAmount - discount;

        repository.save(orderId, finalAmount);
        notifier.notifyCustomer(email);
        auditLogger.log("Order processed successfully: " + orderId);
    }
}
